### ISSUE TITLE
Parental control is failed. Slow AdGuard work (or not working at all)

### ISSUE THREAD

**Issue by alexsannikov** [AdguardTeam/AdGuardHome#1068]
- [ X] I am running the latest version
- [ X] I checked the documentation and found no answer
- [ X] I checked to make sure that this issue has not already been filed

Time to time Internet on my tablet starting to work very slowly.
I see following messages in log:
2019/10/14 11:06:31 [info] Failed to do parental HTTP lookup, ignoring check: Get https://pctrl.adguard.com/check-parental-control-hash?prefixes=C471B89A/19429093/&sensitivity=10: read udp 127.0.0.1:36624->127.0.0.1:53: i/o timeout
2019/10/14 11:06:31 [info] Failed to do parental HTTP lookup, ignoring check: Get https://pctrl.adguard.com/check-parental-control-hash?prefixes=5F9F739A/7932C519/FDF0F7AC/&sensitivity=10: read udp 127.0.0.1:33980->127.0.0.1:53: i/o timeout
2019/10/14 11:06:32 [info] Failed to do parental HTTP lookup, ignoring check: Get https://pctrl.adguard.com/check-parental-control-hash?prefixes=C0A199E7/D455EEF8/&sensitivity=10: read udp 127.0.0.1:55916->127.0.0.1:53: i/o timeout
2019/10/14 11:06:33 [info] Failed to do parental HTTP lookup, ignoring check: Get https://pctrl.adguard.com/check-parental-control-hash?prefixes=2AA39368/D455EEF8/&sensitivity=10: read udp 127.0.0.1:40027->127.0.0.1:53: i/o timeout

After service restart everything works fine again.

* **Version of AdGuard Home server:**
  0.98.1
* **How did you setup DNS configuration:**
Router
* **If it's a router or IoT, please write device model:**
 Raspberry Pi 3 Model B
* **Operating system and version:**
 Raspbian Buster

### Expected Behavior
Work as usual

### Actual Behavior
Works slowly or doesn't work at all

### Additional Information
I use my Raspberry Pi device as a DNS proxy only. It does not pass actual traffic at all.

**Comment by ameshkov** [comment]
Could you please enable verbose logging? This one does not explain what exactly is failing

**Comment by alexsannikov** [comment]
Sure. I will restart service with verbose output and wait till it happen again.
My start line is "/tmp/adguard/AdGuardHome -v -s run"

**Comment by ameshkov** [comment]
Alternatively, you can set log_file and verbose in the config file

**Comment by alexsannikov** [comment]
[adguard.log](https://github.com/AdguardTeam/AdGuardHome/files/3730421/adguard.log)
Have no idea how much this is useful.

**Comment by ameshkov** [comment]
@szolin 

At least this one looks useful:
```
2019/10/14 19:36:10 [info] Failed to do parental HTTP lookup, ignoring check: Get https://pctrl.adguard.com/check-parental-control-hash?prefixes=BE271442/C3B2F630/&sensitivity=10: synthetic.wrap: couldn't dial to 2606:4700:10::6814:1e82:443, cause: dial tcp: address 2606:4700:10::6814:1f82:443: too many colons in address (hidden: dial tcp: address 2606:4700:10::6814:1e82:443: too many colons in address)
```

@alexsannikov 
Please set `verbose` to true in the log file and let's wait till it happens again. So far I don't understand what caused it.

**Comment by alexsannikov** [comment]
Piece of log:
`root@berry(ro): [/tmp/adguard] #  grep "15199#959 " ag_verbose.log
2019/10/18 12:09:30 15199#959 [debug] github.com/AdguardTeam/dnsproxy/proxy.(*Proxy).handleUDPPacket(): Start handling new UDP packet from 192.168.77.201:57237
2019/10/18 12:09:30 15199#959 [debug] github.com/AdguardTeam/dnsproxy/proxy.(*Proxy).logDNSMessage(): IN: ;; opcode: QUERY, status: NOERROR, id: 37408
2019/10/18 12:10:00 15199#959 [debug] github.com/AdguardTeam/AdGuardHome/dnsfilter.(*Dnsfilter).checkParental(): Parental HTTP lookup for www.linkedin.com; Elapsed time: 30186ms
2019/10/18 12:10:00 15199#959 [info] Failed to do parental HTTP lookup, ignoring check: Get https://pctrl.adguard.com/check-parental-control-hash?prefixes=823FB5B2/68C1E55B/&sensitivity=10: read udp 127.0.0.1:39594->127.0.0.1:53: i/o timeout
2019/10/18 12:10:00 15199#959 [debug] github.com/AdguardTeam/dnsproxy/upstream.createDialContext.func1(): Dialing to 1.0.0.1:853
2019/10/18 12:10:00 15199#959 [debug] github.com/AdguardTeam/dnsproxy/upstream.createDialContext.func1(): dialer has successfully initialized connection to 1.0.0.1:853 in 35 milliseconds
2019/10/18 12:11:11 15199#959 [debug] github.com/AdguardTeam/dnsproxy/proxy.exchangeWithUpstream(): upstream 1.0.0.1:853 failed to exchange ;www.linkedin.com.	IN	 A in 70908 milliseconds. Cause: Failed to get a connection from TLSPool to 1.0.0.1:853, cause: Failed to connect to 1.0.0.1, cause: write tcp 192.168.77.99:60620->1.0.0.1:853: i/o timeout
2019/10/18 12:11:11 15199#959 [debug] github.com/AdguardTeam/dnsproxy/upstream.createDialContext.func1(): Dialing to 1.1.1.1:853
2019/10/18 12:11:11 15199#959 [debug] github.com/AdguardTeam/dnsproxy/upstream.createDialContext.func1(): dialer has successfully initialized connection to 1.1.1.1:853 in 44 milliseconds
2019/10/18 12:11:28 15199#959 [debug] github.com/AdguardTeam/dnsproxy/proxy.exchangeWithUpstream(): upstream 1.1.1.1:853 failed to exchange ;www.linkedin.com.	IN	 A in 17386 milliseconds. Cause: Failed to get a connection from TLSPool to 1.1.1.1:853, cause: Failed to connect to 1.1.1.1, cause: write tcp 192.168.77.99:48034->1.1.1.1:853: i/o timeout
2019/10/18 12:11:28 15199#959 [debug] github.com/AdguardTeam/dnsproxy/proxy.(*Proxy).Resolve(): RTT: 88295 ms
2019/10/18 12:11:28 15199#959 [debug] github.com/AdguardTeam/dnsproxy/proxy.(*Proxy).logDNSMessage(): OUT: ;; opcode: QUERY, status: SERVFAIL, id: 37408
2019/10/18 12:11:28 15199#959 [debug] github.com/AdguardTeam/dnsproxy/proxy.(*Proxy).handleUDPPacket(): error handling DNS (udp) request: talking to dnsUpstream failed, cause: all upstreams failed to exchange request, cause: Failed to get a connection from TLSPool to 1.0.0.1:853, cause: Failed to connect to 1.0.0.1, cause: write tcp 192.168.77.99:60620->1.0.0.1:853: i/o timeout (hidden: Failed to get a connection from TLSPool to 1.1.1.1:853, cause: Failed to connect to 1.1.1.1, cause: write tcp 192.168.77.99:48034->1.1.1.1:853: i/o timeout)
root@berry(ro): [/tmp/adguard] #`

How can I upload full log privately?
Thanks

**Comment by alexsannikov** [comment]
Ok, maybe it is not related to parental control, but since some time ago I experience the troubles every day. Log is below.
[adguard.log](https://github.com/AdguardTeam/AdGuardHome/files/3748477/adguard.log)

**Comment by szolin** [comment]
> 2019/10/14 19:36:10 [info] Failed to do parental HTTP lookup, ignoring check: ... (hidden: dial tcp: address 2606:4700:10::6814:1e82:443: too many colons in address)

This error will be fixed with #525 

> Cause: Failed to get a connection from TLSPool to 1.0.0.1:853, cause: Failed to connect to 1.0.0.1, cause: write tcp 192.168.77.99:60620->1.0.0.1:853: i/o timeout

By looking at your log file I can see that you were experiencing network problems for about a minute.  Your requests to 1.0.0.1:853 took 20-30 seconds - they are timed out.  You should probably use another upstream server or fix your network connection.
